From 5332d8432cf1a751cb60793ba8e80e7540dce9c0 Mon Sep 17 00:00:00 2001
From: Max Reitz <mreitz@redhat.com>
Date: Mon, 4 Nov 2013 22:32:16 +0100
Subject: [PATCH 23/87] block/raw: Employ error parameter

RH-Author: Max Reitz <mreitz@redhat.com>
Message-id: <1383604354-12743-26-git-send-email-mreitz@redhat.com>
Patchwork-id: 55325
O-Subject: [RHEL-7.0 qemu-kvm PATCH 25/43] block/raw: Employ error parameter
Bugzilla: 1026524
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Laszlo Ersek <lersek@redhat.com>
RH-Acked-by: Fam Zheng <famz@redhat.com>
RH-Acked-by: Stefan Hajnoczi <stefanha@redhat.com>

BZ: 1026524

Propagate errors in raw_create rather than directly reporting and
afterwards discarding them.

This downstream-only commit replaces upstream
92f1deec317230575726a8e0ab5c110781d30ec0 (which did the same for
raw_bsd.c instead of raw.c).

Signed-off-by: Max Reitz <mreitz@redhat.com>
---
 block/raw.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

Signed-off-by: Miroslav Rezanina <mrezanin@redhat.com>
---
 block/raw.c |    3 +--
 1 files changed, 1 insertions(+), 2 deletions(-)

diff --git a/block/raw.c b/block/raw.c
index 50073b6..66eda91 100644
--- a/block/raw.c
+++ b/block/raw.c
@@ -113,8 +113,7 @@ static int raw_create(const char *filename, QEMUOptionParameter *options,
 
     ret = bdrv_create_file(filename, options, &local_err);
     if (error_is_set(&local_err)) {
-        qerror_report_err(local_err);
-        error_free(local_err);
+        error_propagate(errp, local_err);
     }
     return ret;
 }
-- 
1.7.1

